plugins/modules/jdiff.py

```diff
@@ -0,0 +1,228 @@
+  jdiff:
+    check_type: "tolerance"
+    evaluate_args:
+      reference_data: "{{ ref_data }}"
+      value_to_compare: "{{ data_to_compare }}"
+      tolerance: 5
+    jmespath: "*.*.ipv4.[accepted_prefixes]"
+  register: result
+  vars:
+    ref_data: |
+      {
+          "10.1.0.0": {
+              "address_family": {
+                  "ipv4": {
+                      "accepted_prefixes": 100,
+                      "sent_prefixes": 1
+                  }
+              }
+          }
+      }
+    data_to_compare: |
+      {
+          "10.1.0.0": {
+              "address_family": {
+                  "ipv4": {
+                      "accepted_prefixes": 90,
+                      "sent_prefixes": 0
+                  }
+              }
+          }
+      }
+
+- name: "PARAMETER - VALIDATE PEER TYPE"
+  jdiff:
+    check_type: "parameter_match"
+    evaluate_args:
+      value_to_compare: "{{ data_to_compare }}"
+      mode: match
+      params:
+        linkType: external
+    jmespath: peers[*].[$ip$,linkType]
+  register: result
+  vars:
+    data_to_compare: |
+      {
+          "peers": [
+              {
+                  "ip": "10.1.0.0",
+                  "linkType": "external"
+              },
+              {
+                  "ip": "10.2.0.0",
+                  "linkType": "external"
+              }
+          ]
+      }
+
+- name: "REGEX - VALIDATE MAC FORMAT"
+  jdiff:
+    check_type: "regex"
+    evaluate_args:
+      value_to_compare: "{{ data_to_compare }}"
+      regex: "^([0-9a-fA-F]{2}(:|-)){5}([0-9a-fA-F]{2})$"
+      mode: match
+    jmespath: interfaces.*.[$name$,burnedInAddress]
+  register: result
+  vars:
+    data_to_compare: |
+      {
+          "interfaces": {
+              "Management1": {
+                  "burnedInAddress": "08:00:27:e6:b2:f8",
+                  "name": "Management1"
+              },
+              "Management2": {
+                  "burnedInAddress": "08-00-27-e6-b2-f9",
+                  "name": "Management2"
+              }
+          }
+      }
+
+- name: "OPERATOR - VALIDATE RX LEVEL WITHIN RANGE"
+  jdiff:
+    check_type: "operator"
+    evaluate_args:
+      value_to_compare: "{{ data_to_compare }}"
+      params:
+        params:
+          mode: in-range
+          operator_data: [-8, -2]
+    jmespath: ports[*].[$name$,RxPower]
+  register: result
+  vars:
+    data_to_compare: |
+      {
+          "ports": [
+              {
+                  "name": "1/1",
+                  "RxPower": -3.83,
+                  "state": "Connected"
+              },
+              {
+                  "name": "1/2",
+                  "RxPower": -3.21,
+                  "state": "Connected"
+              }
+          ]
+      }
+"""
+
+RETURN = """
+changed:
+    description: Indicates if change was made - always False.
+    returned: success
+    type: bool
+fail_details:
+    description: deepdiff output indicating where the check failed
```

Comment:
do we need to expose our backend library? maybe a more generic explanation would save us explanations in the future